tasks.py

```diff
@@ -1,76 +1,77 @@
 from invoke import task
 from invoke.exceptions import Failure
+from invoke.tasks import call
+
+
+def run_or_fail(c, cmd, fail_message="Error", fail_callback=None):
+    try:
+        cmd(c)
+    except Failure as ex:
+        print(fail_message)
+        if fail_callback:
+            fail_callback(c)
+        exit()
 
 @task()
+def tools_check(c, tools=[]):
+    missing_tools = []
+    for tool in tools:
+        if c.run(f"which {tool}", hide=True, warn=True).exited != 0:
+            missing_tools.append(f"{tool}")
+
+    if len(missing_tools) > 0:
+        print("These tools are required : {}".format(", ".join(missing_tools)))
+        print("Please install them")
+        exit()
```

Comment:
@patrick91 I used the approach you recommended to me, but I envisaged the possibility of passing the tools to be checked by the task so that it could be used as a prerequisite for other tasks.

I hope you like it :D

Reply:
I like this tools check ✨

Reply:
The only issue I see is in case any tool, for some reason, do not work with `which` but I we can handle that in the future